### PR TITLE
Adding a pull request action to lint files.

### PR DESCRIPTION
**Pull request by ryanwelcher**
Part of #172, this PR introduces a pull request action to lint all files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
 		"@wordpress/env": "^4.6.0",
 		"@wordpress/scripts": "^23.0.0",
 		"copy-webpack-plugin": "^10.2.4",
+		"eslint-plugin-prettier": "^4.0.0",
 		"prettier": "npm:wp-prettier@2.2.1-beta-1"
```
Comment (gziolo) [inline package.json:23]: It's super annoying we need to add them explicitly 😞 

I had to do the same trick in my personal projects.

Comment (ryanwelcher) [reply package.json:23]: Yes, maybe this is just part of the best practice until we can use the standard prettier 😞

Comment (gziolo) [reply package.json:23]: maybe we should also allow `*` for prettier when configuring it as a peer dependency

Comment (ryanwelcher) [reply package.json:23]: Like this - `"prettier": "npm:wp-prettier@*"`?

File: .github/workflows/pull-request-actions.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Static Linting
+
+on: [pull_request]
```
Comment (gziolo) [inline .github/workflows/pull-request-actions.yml]: Should it also trigger when merged to the main branch?

**Review by gziolo — APPROVED** [review]
Looks great and everything passes. Let’s investigate Node.js 18 separately.

File: .github/workflows/pull-request-actions.yml (hunk 2 above)
Comment (ryanwelcher) [reply .github/workflows/pull-request-actions.yml]: That's not a bad idea, I'll add it before merging. Thanks!

File: package.json (hunk 1 above)
Comment (gziolo) [reply package.json:23]: Not here, in Gutenberg if that’s possible.

**Pull request merged**